geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaUtils.scala

```diff
@@ -0,0 +1,150 @@
+                                versionId: Short,
+                                correlationId: Int,
+                                clientId: String): OffsetCommitRequest
+}
+
+object KafkaUtilsLoader extends LazyLogging {
+  lazy val kafkaUtils: AbstractKafkaUtils = {
+    val kuIter = ServiceLoader.load(classOf[AbstractKafkaUtils]).iterator()
+    if (kuIter.hasNext) {
+      val first = kuIter.next()
+      if (kuIter.hasNext) {
+        logger.warn(s"Multiple geomesa KafkaUtils found.  Should only have one. Using the first: '$first'")
+      }
+      first
+    } else {
+      logger.debug(s"No geomesa KafkaUtils found.  Using default one for 0.8.")
+      DefaultKafkaUtils
+    }
+  }
+}
+
+/**
+  * Default KafkaUtils for kafka 0.8
+  */
+object DefaultKafkaUtils extends AbstractKafkaUtils {
+  def channelToPayload: (BlockingChannel) => ByteBuffer = _.receive().buffer
+  def channelSend(bc: BlockingChannel, requestOrResponse: RequestOrResponse): Long = bc.send(requestOrResponse).toLong
+  def leaderBrokerForPartition: PartitionMetadata => Option[Broker] = _.leader.map(l => Broker(l.host, l.port))
+  def assign(partitionAssignor: PartitionAssignor, ac: AssignmentContext) = partitionAssignor.assign(ac)
+  def createZkUtils(zkConnect: String, sessionTimeout: Int, connectTimeout: Int): AbstractZkUtils = {
+    // zkStringSerializer is required - otherwise topics won't be created correctly
+    DefaultZkUtils(new ZkClient(zkConnect, sessionTimeout, connectTimeout, ZKStringSerializer))
+  }
+  def tryFindNewLeader(tap: TopicAndPartition,
+                       partitions: Option[Seq[PartitionMetadata]],
+                       oldLeader: Option[Broker],
+                       tries: Int): Option[Broker] = {
+    val maybeLeader = partitions.flatMap(_.find(_.partitionId == tap.partition)).flatMap(_.leader)
+    val leader = oldLeader match {
+      // first time through if the leader hasn't changed give ZooKeeper a second to recover
+      // second time, assume the broker did recover before failover, or it was a non-Broker issue
+      case Some(old) => maybeLeader.filter(m => (m.host != old.host && m.port != old.port) || tries > 1)
+      case None      => maybeLeader
+    }
+
+    leader.map(l => Broker(l.host, l.port))
+  }
+  def rm(file: File): Unit = Utils.rm(file)
+  def createOffsetAndMetadata(offset: Long, time: Long): OffsetAndMetadata = OffsetAndMetadata(offset, timestamp = time)
+  def createOffsetCommitRequest(groupId: String,
+                                requestInfo: immutable.Map[TopicAndPartition, OffsetAndMetadata],
+                                versionId: Short,
+                                correlationId: Int,
+                                clientId: String): OffsetCommitRequest =
+    new OffsetCommitRequest(groupId, requestInfo, versionId, correlationId, clientId)
+}
+
+/**
+  * Default ZkUtils for kafka 0.8
+  */
+case class DefaultZkUtils(zkClient: ZkClient) extends AbstractZkUtils {
```

Comment:
[minor] I'd pull out the AbstractZkUitls and DeafultZkUtils into a separate file.